pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -149,16 +149,21 @@ func (c *defaultSeedRegistrationControl) Reconcile(shootObj *gardencorev1beta1.S
 	}
 
 	if shoot.DeletionTimestamp == nil && shootedSeedConfig != nil {
-		if shoot.Status.LastOperation == nil || shoot.Status.LastOperation.State != gardencorev1beta1.LastOperationStateSucceeded {
+		if shoot.Generation != shoot.Status.ObservedGeneration || shoot.Status.LastOperation == nil || shoot.Status.LastOperation.State != gardencorev1beta1.LastOperationStateSucceeded {
 			shootLogger.Infof("[SHOOTED SEED REGISTRATION] Waiting for shoot %s to be reconciled before registering it as seed", shoot.Name)
 			return reconcile.Result{
 				RequeueAfter: 10 * time.Second,
 			}, nil
 		}
 
+		seedClient, err := c.clientMap.GetClient(ctx, keys.ForSeedWithName(*shootObj.Spec.SeedName))
```

Comment:
Hmm, I'm wondering if `shootObj.Spec.SeedName` can be `nil` here.
This is implicitly covered by `shoot.Generation != shoot.Status.ObservedGeneration`, right?

Reply:
Probably it cannot happen because a) the events [are filtered](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/shoot/shoot.go#L132) to match the gardenlet's seed config, (i.e., when the `Shoot` has no `.spec.seedName` it will never reach this code path), and b) the `.status.lastOperation` will only be set if the `Shoot` is being reconciled (which only happens when it's assigned to a `Seed`).

Reply:
Ah, right. Great! :)